tests/core/data_storage/test_on_disk_storage.py

```diff
@@ -0,0 +1,174 @@
+import functools
+
+import numpy
+
+from openfisca_core import data_storage
+from openfisca_core import periods
+
+import pytest
+
+
+@pytest.fixture
+def storage():
+    return functools.partial(data_storage.OnDiskStorage, storage_dir = "/tmp")
+
+
+@pytest.fixture
+def eternal_storage(storage):
+    return functools.partial(storage, is_eternal = True)
+
+
+@pytest.fixture
+def period():
+    return periods.period("2020")
+
+
+@pytest.fixture
+def eternal_period():
+    return periods.period(periods.ETERNITY)
+
+
+@pytest.fixture
+def value():
+    return numpy.array([1])
+
+
+@pytest.fixture
+def file(storage, period):
+    return f"{storage().storage_dir}/{period}.npy"
+
+
+@pytest.fixture
+def eternal_file(storage, eternal_period):
+    return f"{storage().storage_dir}/{str(eternal_period)}.npy"
+
+
+def test___init__(storage):
+    result = storage()
+
+    assert result
+
+
+def test___init__when_is_eternal(eternal_storage):
+    result = eternal_storage()
+
+    assert result.is_eternal
+
+
+def test__init__when_preserve_storage_dir(storage):
+    result = storage(preserve_storage_dir = True)
+
+    assert result.preserve_storage_dir
+
+
+def test_get(storage, period, file, mocker):
+    storage = storage()
+    files = {period: file}
+    mocker.patch.dict(storage._files, files)
+    mocker.patch.object(storage, "_decode_file", autospec = True)
+    storage.get(period)
+
+    result = files.get(period)
+
+    storage._decode_file.assert_called_once_with(result)
```

Comment:
It's really difficult for me to make any sense of this test. I don't have any intuitive sense of what is being guaranteed. I have no idea what `autospec` does or why it's relevant here. The test demands detailed knowledge of the implementation and thus fails to provide much value as documentation.

Reply:
>  I don't have any intuitive sense of what is being guaranteed.

I'm trying to separate the caching logic with the data I/O logic, so the intention is to guarantee that, given the expected conditions, responsibility will be handed over from the caching layer to the disk I/O layer.

I'm not really interested on what happens after caching is done, so I'm just guaranteeing the correct message is being handed. That'll help us change the caching contract without affecting the data I/O one.

>  I have no idea what autospec does or why it's relevant here.

It makes sure you're respecting the actual contract of the method you're mocking. Which is exactly what we want : we can to make sure that three conditions are respected :

- when responsibility is handed over
- arity
- type

If those 3 are respected, I feel confident going and tinkering around with the caching layer.

> The test demands detailed knowledge of the implementation and thus fails to provide much value as documentation.

That's correct, do you prefer if I add some docstring to the test itself ? All this is a roadmap to get rid of these classes.

Reply:
I'm reworking the API otherwise those tests won't make sense, major bump unavoidable ...

Reply:
I changed of approach radically, so this code is no longer updated.